e2e/tests/ssm.test.js

```diff
@@ -216,5 +217,95 @@ describe('ssm', async () => {
       expect(result).to.not.equal(undefined)
       expect(result.body.status.status).to.contain('namespace does not allow to assume role let-me-be-root')
     })
+    it('should not pull from ssm after waiting POLLER_INTERVAL_MILLISECONDS', async () => {
+      backends.systemManager.checkUpdated = true
+
+      backends.systemManager.intervalPolling = 1000
+
+      let result = await putParameter({
+        Name: `/e2e/ssmcheck/${uuid}`,
+        Type: 'String',
+        Value: 'foo'
+      }).catch(err => {
+        expect(err).to.equal(null)
+      })
+
+      result = await kubeClient
+        .apis[customResourceManifest.spec.group]
+        .v1.namespaces('default')[customResourceManifest.spec.names.plural]
+        .post({
+          body: {
+            apiVersion: 'kubernetes-client.io/v1',
+            kind: 'ExternalSecret',
+            metadata: {
+              name: `e2e-ssm-ssmcheck-${uuid}`
+            },
+            spec: {
+              backendType: 'systemManager',
+              data: [
+                {
+                  key: `/e2e/ssmcheck/${uuid}`,
+                  name: 'name'
+                }
+              ]
+            }
+          }
+        })
+
+      await new Promise(resolve => setTimeout(resolve, 5000))
+
+      expect(result).to.not.equal(undefined)
+      expect(result.statusCode).to.equal(201)
+
+      result = await kubeClient
+        .apis[customResourceManifest.spec.group]
+        .v1.namespaces('default')
+        .externalsecrets(`e2e-ssm-ssmcheck-${uuid}`)
+        .get()
+      expect(result).to.not.equal(undefined)
+      expect(result.body.status.status).to.contain('The parameter has not changed since last time')
+    })
+    it('should pull from ssm with 500 history of parameters', async () => {
+      backends.systemManager.checkUpdated = true
+
+      for (let i = 1; i <= 500; i++) {
+        await putParameter({
+          Name: `/e2e/ssmcheck/${uuid}`,
```

Comment:
To properly test the parameter history pagination and in-memory caching behavior with many many parameters you would need to variablize the name to, e.g. `/e2e/ssmcheck/${uuid}/${i}` instead of overriding the value.